app/src/test/java/com/example/campusguide/utils/DatabaseTest.kt

```diff
@@ -0,0 +1,73 @@
+package com.example.campusguide.utils
+
+import database.ObjectBox
+import database.entity.*
+import io.objectbox.BoxStore
+import io.objectbox.DebugFlags
+import io.objectbox.kotlin.boxFor
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import org.junit.Assert.assertEquals as assertEquals
+
+class DatabaseTest {
+    private val TEST_DIRECTORY: File = File("test-db/objectbox/")
+    private var store: BoxStore? = null
+
+    @Before
+    fun setUp() { // delete database files before each test to start with a clean database
+        BoxStore.deleteAllFiles(TEST_DIRECTORY)
+        store =
+            MyObjectBox.builder() // add directory flag to change where ObjectBox puts its database files
+                .directory(TEST_DIRECTORY) // optional: add debug flags for more detailed ObjectBox log output
+                .debugFlags(DebugFlags.LOG_QUERIES or DebugFlags.LOG_QUERY_PARAMETERS)
+                .build()
+    }
+
+    @After
+    fun tearDown() {
+        store!!.close()
+        store!!.deleteAllFiles()
+    }
+
+    @Test
+    fun whenAddABuilding_ThenShouldBeInDatabse(){
+        //given
+        val store = store!!.boxFor<Building>()
+        val outlinePoint = Point(1, 10.0, 20.0)
+        val holePoint = Point(2, 30.0, 40.0)
+        val building  = Building(0, "Name", "abbreviation")
+        building.highlight.target = Highlight(0)
+        val outline = Outline(0)
+        outline.points.add(outlinePoint)
+        val hole = Hole(0)
+        hole.points.add(holePoint)
+        outline.holes.add(hole)
+        building.highlight.target.outlines.add(outline)
+
+        //when
+        store.put(building)
+
+        //then
+        val buildings = store.query().equal(Building_.fullName, "Name").build().find()
+        assertEquals(1, buildings.size)
+        val dbBuilding = buildings[0]
+        assertEquals(dbBuilding.fullName, "Name")
+        assertEquals(dbBuilding.abbreviationName, "abbreviation")
+
+        assertEquals(dbBuilding.highlight.target.outlines.size, 1)
+        val dbOutline = dbBuilding.highlight.target.outlines[0]
+        assertEquals(dbOutline.points.size, 1)
+        assertEquals(dbOutline.points[0].order,1)
+        assertEquals(dbOutline.points[0].latitude,10.0, 0.00001)
```

Comment:
I'm assuming 0.00001 is the tolerance because these are floating point numbers? In that case also stick this value in a constant so the tolerance can easily be adjusted

Reply:
Also, put your hardcoded numbers like 10.0 into variables that are used in the object creation and in the assertions.